mp4/senc.go

```diff
@@ -116,15 +131,36 @@ func DecodeSenc(hdr BoxHeader, startPos uint64, r io.Reader) (Box, error) {
 
 // DecodeSencSR - box-specific decode
 func DecodeSencSR(hdr BoxHeader, startPos uint64, sr bits.SliceReader) (Box, error) {
+	if hdr.Size < 16 {
+		return nil, fmt.Errorf("box size %d less than min size 16", hdr.Size)
+	}
+
 	versionAndFlags := sr.ReadUint32()
+	version := byte(versionAndFlags >> 24)
+	if version > 0 {
+		return nil, fmt.Errorf("version %d not supported", version)
+	}
+
+	if version > 0 {
+		return nil, fmt.Errorf("version %d not supported", version)
+	}
```

Comment:
Looks like you accidentally duplicated this condition.

Reply:
ok. thanks for pointing that out.